EasyPost/ReferralCustomer.cs

```diff
@@ -0,0 +1,192 @@
+        }
+
+        /// <summary>
+        ///     Update a Referral Customer object email.
+        ///     This function requires the Partner User's API key.
+        /// </summary>
+        /// <param name="email">Email of the referral user to update.</param>
+        /// <param name="userId">ID of the referral user to update.</param>
+        /// <returns>True if update was successful, false otherwise.</returns>
+        public static async Task<bool> UpdateEmail(string email, string userId)
+        {
+            Dictionary<string, object> wrappedParams = new Dictionary<string, object>
+            {
+                {
+                    "user", new Dictionary<string, object>
+                    {
+                        {
+                            "email", email
+                        }
+                    }
+                }
+            };
+
+            Request request = new Request($"referral_customers/{userId}", Method.Put);
+            request.AddParameters(wrappedParams);
+
+            return await request.Execute();
+        }
+
+        /// <summary>
+        ///     List all Referral Customer objects.
+        ///     This function requires the Partner User's API key.
+        /// </summary>
+        /// <param name="parameters">Parameters for API call.</param>
+        /// <returns>An EasyPost.ReferralCustomerCollection instance.</returns>
+        public static async Task<ReferralCustomerCollection> All(Dictionary<string, object>? parameters = null)
+        {
+            Request request = new Request("referral_customers", Method.Get);
+            request.AddParameters(parameters);
+
+            return await request.Execute<ReferralCustomerCollection>();
+        }
+
+        /// <summary>
+        ///     Add a credit card to a Referral Customer.
+        ///     This function requires the Referral User's API key.
+        /// </summary>
+        /// <param name="referralApiKey">API key of the referral user.</param>
+        /// <param name="number">Credit card number.</param>
+        /// <param name="expirationMonth">Expiration month of the credit card.</param>
+        /// <param name="expirationYear">Expiration year of the credit card.</param>
+        /// <param name="cvc">CVC of the credit card.</param>
+        /// <param name="priority">Priority of the credit card.</param>
+        /// <returns>An EasyPost.PaymentMethodObject instance.</returns>
+        /// <exception cref="Exception">When the request fails.</exception>
+        public static async Task<PaymentMethodObject> AddCreditCardToUser(string referralApiKey, string number, int expirationMonth, int expirationYear, string cvc, PaymentMethod.Priority? priority = null)
+        {
+            string easypostStripeApiKey = await RetrieveEasypostStripeApiKey();
+            string stripeToken;
+
+            try
+            {
+                stripeToken = await CreateStripeToken(number, expirationMonth, expirationYear, cvc, easypostStripeApiKey);
```

Comment:
This variable should have a fallback if we cannot get it from Stripe (in other libs, we set it to an empty string then check for existence), this is important since we don't control this endpoint and they could (although unlikely) change the expectations of this response.

Reply:
That check is done at the end of the `CreateStripeToken` function.